### PR TITLE
truncate and scroll list item text (and some misc...)

### PR DESCRIPTION
**Pull request by HookedBehemoth**
The last commit message doesn't make sense because I have it as a sub-repo. ^^

### CODE REVIEW

File: include/tesla.hpp [hunk 1]
```diff
@@ -644,6 +659,11 @@ namespace tsl {
                 u32 i = 0;
 
                 do {
+                    if (maxWidth > 0 && maxWidth < (currX - x))
+                        break;
+
+                    if (written) *written += 1;
```
Comment (WerWolv) [inline include/tesla.hpp]: Should be `if (written != nullptr)` to match the rest of the code

File: include/tesla.hpp [hunk 2]
```diff
@@ -1325,27 +1345,73 @@ namespace tsl {
              * 
              * @param text Initial description text
              */
-            ListItem(std::string text) : Element(), m_text(text) {}
+            ListItem(const std::string& text) : Element(), m_text(text), m_value(""), m_scrollText(""), m_ellipsisText(""), m_scroll(), m_trunctuated(), m_faint(), m_maxScroll(), m_scrollOffset(), m_maxWidth(), m_counter() {
```
Comment (WerWolv) [inline include/tesla.hpp]: Split this up into multiple lines. It's way way too long

File: include/tesla.hpp [hunk 3]
```diff
@@ -1325,27 +1345,73 @@ namespace tsl {
...
 
-                renderer->drawString(this->m_value.c_str(), false, this->getX() + this->getWidth() - this->m_valueWidth - 20, this->getY() + 45, 20, this->m_faint ? a({ 0x6, 0x6, 0x6, 0xF }) : a({ 0x5, 0xC, 0xA, 0xF }));
+                renderer->drawString(this->m_value.c_str(), false, this->getX() + this->m_maxWidth + 45, this->getY() + 45, 20, this->m_faint ? a({ 0x6, 0x6, 0x6, 0xF }) : a({ 0x5, 0xC, 0xA, 0xF }));
             }
 
             virtual void layout(u16 parentX, u16 parentY, u16 parentWidth, u16 parentHeight) override {
                 
             }
 
+            virtual void setFocused(bool state) final {
```
Comment (WerWolv) [inline include/tesla.hpp]: This should probably not be final

**Review by WerWolv — CHANGES_REQUESTED** [review]

**Comment by HookedBehemoth** [comment]
I reverted setting defaults in the constructor as you don't do it anywhere else in the code.

**Pull request merged**